torch/distributed/_tensor/random.py

```diff
@@ -361,11 +363,15 @@ def _distribute_region(self, spec: DTensorSpec):
             )
 
         if self.distribute_region_enabled:
-            with torch.random.fork_rng(self._devices):
-                torch.cuda.set_rng_state(self.rng_states["tensor-parallel-rng"])
+            with torch.random.fork_rng(self._devices, device_type=self._device_type):
```

Comment:
same here, we can't really pass in `device_type` to this fork_rng?

Reply:
emm, actually in [source code on github](https://github.com/pytorch/pytorch/blob/main/torch/random.py), `fork_rng` accept a `device_type` arg. Also, I have built my pytorch project and run tests with it. It turns out that it's necessary to pass the `device_type` to support third-party devices.